PostHog/Autocapture/PostHogAutocaptureEventTracker.swift

```diff
@@ -363,9 +369,35 @@
         }
     }
 
+    private func typeName(of type: AnyClass) -> String {
+        let typeName = String(describing: type)
+        if let match = typeName.range(of: "^[^<]+", options: .regularExpression) {
+            // Extracts everything before the first '<' to deal with generics
+            return String(typeName[match])
+        }
+        return typeName
+    }
+
     extension NSObject {
         var descriptiveTypeName: String {
-            String(describing: type(of: self))
+            typeName(of: type(of: self))
+        }
+
+        var baseTypeNames: String {
+            var baseTypes: [String] = []
+            var previousType: AnyClass = type(of: self)
+
+            // traverse the inheritance tree
+            while let superclass = previousType.superclass() {
+                // NSObject or UIResponder are base types for almost all UIKit and not really helpful here
+                if superclass == NSObject.self || superclass == UIResponder.self {
+                    return baseTypes.joined(separator: " ")
+                }
+                previousType = superclass
+                baseTypes.append(typeName(of: superclass))
+            }
+
+            return baseTypes.joined(separator: " ")
```

Comment:
Current example:

`<UITextField id="usernameTextField" class="UIControl UIView">I can be labeled</UITextField>`

I'm not convinced the `class` with nested hierarchy is useful. I just can't think of a useful example, do we use that at PostHog or can you think of an insight that would require the hierarchy tree? Maybe just the parent's class is fine if it's not the same type eg:

 `<UITextField id="usernameTextField" class="UIControl">I can be labeled</UITextField>`

But having the duplicated "UIView" and friends just seems wasting bytes over the network, happy to be wrong here, just need to think if its useful

Reply:
yeah just superclass would be fine, I think the web makes sense to be more chatty because its everything a `div` so they need more info, it's different for us.
Maybe add a maxDepth for super classes as well so we don't have super long nested base classes, and gain some perf improvements in there by traversing the list a bit less